### PR TITLE
Modernization

### PR DESCRIPTION
**Pull request by CodeFusion**
- Update libraries to their newest versions
- Change targetAPI to 31
- Migrate kotlinx synthetics to view binding
- Additional small improvements

### CODE REVIEW

**Comment by hofmanntm** [comment]
I think Romulo should do this review. And likely should wait until the other PR is merged :)

File: build.gradle
Comment (khaled-Mindsea) [inline build.gradle:32]: should we also update the target to 31?

**Review by ghost — APPROVED** [review]
LGTM! 
can we add those .idea files to gitIgnore ?

**Pull request merged**
